aptos-move/framework/aptos-framework/sources/transaction_fee.move

```diff
@@ -19,9 +108,193 @@ module aptos_framework::transaction_fee {
         );
     }
 
+    /// Burn a collected transaction fee.
+    public(friend) fun burn_collected_fee(coin: Coin<AptosCoin>) acquires AptosCoinCapabilities {
+        if (coin::value(&coin) == 0) {
```

Comment:
Is this check necessary?

Reply:
Yes, apparently you cannot burn a zero-valued coin and should use `destroy_zero`. Zero-valued coin appears when we burn 100% of coin (so 0 is left)

Reply:
Saw the comment above, having `burn_coin_fraction` should help to avoid this actually.